### ISSUE TITLE
Electron - For the module field, exclude anything before .asar/

### ISSUE THREAD

**Issue by armenzg** [getsentry/sentry-electron#686]
I'm noticing Electron stacktraces including a full path for the `module` value rather than just the module.
For instance, this: `C:\Users\User\AppData\Local\Programs\<company_name_here>\resources\app.asar\main` instead of just `main` by replacing everything that comes before `.asar\`.

The data I'm looking at may have no source maps or lack of context lines.

This causes grouping issues downstream. You can validate this in an Electron project with Discover by querying for `stack.module:"C:\Users*" ` and using the following columns (see screenshot).
<img width="475" alt="image" src="https://github.com/getsentry/sentry-electron/assets/44410/26f71e2a-bd86-4a69-a165-e5f7231fc704">

I see the following values:
* module: `C:\\Users\\User\\AppData\\Local\\Programs\\<company_name_here>\\resources\\app.asar\\main
* filename: `app:///main.js`
* abs_path: `app:///main.js`

I would expect the values to be:
* module: `main`
* filename: `app:///main.js`
* abs_path: `C:\\Users\\User\\AppData\\Local\\Programs\\<company_name_here>\\resources\\app.asar\\main.js`

Q: Would this be a reasonable fix?

**Comment by armenzg** [comment]
From [the original issue](https://github.com/getsentry/sentry-javascript/issues/8205):

`module` in stacktraces include the user account's installation path, thus, causing groups to be created per user.

This is snippet is from project 5599205 and group 4019798020 (redacted name of user):
```
{
"function": "IncomingMessage.<anonymous>",
"module": "C:\\Users\\NameOfUser\\AppData\\Local\\Programs\\AppName\\resources\\app.asar\\main",
"filename": "app:///main.js",
"abs_path": "app:///main.js",
"lineno": 449,
"colno": 544,
```

If you use Discover and filter for `title:"HttpError: 403 Forbidden"` and group per unique issues you will see that 35k groups were created for 11M events in the last 90 days.

![Image](https://github.com/getsentry/sentry-javascript/assets/44410/26523c1a-8974-45ef-893c-8b3cd49f4c8b)

**Comment by lforst** [comment]
Hi, the project you shared uses quite an old version of the electron SDK (`2.5.4`). The current version of the SDK already normalizes/strips away user paths for grouping and PII reasons.

**Comment by timfish** [comment]
> You can validate this in an Electron project with Discover by querying for `stack.module:"C:\Users*"`

I've looked through my production apps and can't find anything so I would definitely try updating to the SDK to the latest first.

> The current version of the SDK already normalizes/strips away user paths for grouping and PII reasons

It should but this might be something we're missing in some cases with `module`. We don't attempt to normalise that since I've never seen it containing absolute paths!

@armenzg, if you're still getting issues with the latest SDK, can you supply the full, raw, minified stack string (with username redacted) and I should be able to get a better idea of what's going on. You can get that by ticking these two boxes:

<img width="202" alt="image" src="https://github.com/getsentry/sentry-electron/assets/1150298/5fbecd3e-978a-4542-9c0e-27e38ff3b8a7">

**Comment by armenzg** [comment]
@timfish I'm reporting on behalf of the customer, thus, I can't upgrade their SDK.

I've got the analytics showing how many groups got created on June 1st by SDK version and project ID.
You can see the spreadsheet [here](https://docs.google.com/spreadsheets/d/1U4R_oZUhViU3qud8Q0BpqLtZqhAFNdrQLNGOQzjJJgE/edit#gid=54845231).

Snapshot:
<img width="404" alt="image" src="https://github.com/getsentry/sentry-electron/assets/44410/b90343e7-8cb0-458a-92f4-ef996b0c2e11">

Query:
```
SELECT
  sdk_name, sdk_version, project_id,
  COUNT(DISTINCT group_id) AS cnt
FROM `super-secure-big-data.streams_partitioned.errors_gcs`,
UNNEST(exception_frames_module) AS m,
UNNEST(contexts_key) AS ckey WITH OFFSET c1,
UNNEST(contexts_value) AS cvalue WITH OFFSET c2
WHERE dt = '2023-06-01'AND 
      c1 = c2 AND 
      ckey = "runtime.name" AND cvalue = "Electron" AND STARTS_WITH(m, "C:\\Users")
GROUP BY 1,2,3
ORDER BY 4 DESC
```

**Comment by timfish** [comment]
@armenzg are you able to supply an example of the full, raw, minified stack string with username redacted?

**Comment by armenzg** [comment]
@timfish Does this help? Thanks for looking into this!
I wonder if the value of the culprit is used as part of the module when the file is `app:///main.js`

```
HttpError: 403 Forbidden
"method: GET url: https://s3-us-west-2.amazonaws.com/<redacted>/desktop-packages/latest.yml?noCache=1h1mini3k\n\n          Data:\n          <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDe...
  File "app:///main.js", line 427, col 4694, in s
    {snip} 41659),n=c.default("electron-builder");function s(_,T=null){return new r(_.statusCode||-1,`${_.statusCode} ${_.statusMessage}`+(T==null?"":`
  File "app:///main.js", line 432, col 544, in IncomingMessage.<anonymous>
    {snip} .find(Ne=>Ne.includes("json"))!=null:J.includes("json"));I(s(T,`method: ${O.method||"GET"} url: ${O.protocol||"https:"}//${O.hostname}${O.po {snip}
  File "events.js", line 315, col 20, in IncomingMessage.emit
  File "domain.js", line 467, col 12, in IncomingMessage.EventEmitter.emit
  File "internal/streams/readable.js", line 1327, col 12, in endReadableNT
  File "internal/process/task_queues.js", line 80, col 21, in processTicksAndRejections
```


```json
{
"culprit": "s(C:\\Users\\regin\\AppData\\Local\\Programs\\Loom\\resources\\app.asar\\main)",
"exception": {
    "values": [
    {
    "stacktrace": {
        "frames": [
            {
            "function": "IncomingMessage.<anonymous>",
            "module": "C:\\Users\\<redacted>\\AppData\\Local\\Programs\\<redacted>\\resources\\app.asar\\main",
            "filename": "app:///main.js",
            "abs_path": "app:///main.js",
            "lineno": 432,
            "colno": 544,
            "pre_context": [
                "{snip} efineProperty(w,\"__esModule\",{value:!0}),w.CancellationError=w.CancellationToken=void 0;const e=t(82            361);class c extends e.EventEmitter{cons {snip}",
                "`+JSON.stringify(T,null,\"  \"))+`",
                "{snip} tusCode=T,this.description=N,this.name=\"HttpError\",this.code=`HTTP_ERROR_${T}`}isServerError(){return this.statusCode>=500&&this.statusCode< {snip}",
                "",
                "Please double check that your authentication token is correct. Due to security reasons, actual status maybe not reported, but 404."
            ],
            "context_line": "{snip} .find(Ne=>Ne.includes(\"json\"))!=null:J.includes(\"json\"));I(s(T,`method: ${O.method||\"GET\"} url: ${O.protocol||\"https:\"}//${O.hostname}${O.po {snip}",
            "post_context": [
                "",
                "          Data:",
                "          ${_e?JSON.stringify(JSON.parse(G)):G}",
                "{snip} owed size is 500 MB\"));return}F=Buffer.alloc(G),K=0}}V.on(\"data\",G=>{if(K!==-1)G.copy(F,K),K+=G.length;else if(F==null)F=G;else{if(F.length> {snip}",
                "`).join(`"
            ],
            "in_app": true
},
```

**Comment by timfish** [comment]
I've run some tests with the [`getModule` function](https://github.com/getsentry/sentry-javascript/blob/develop/packages/node/src/module.ts#L13) that gets the module name from a filename and it correctly parses the above paths so I'm not sure what's causing this yet.

**Comment by timfish** [comment]
I still have no idea what's causing this but you can disable the module parsing for stack frames in the main process by overriding the `stackParser` option with the node parser but without module parsing:

```ts
import { createStackParser, nodeStackLineParser } from '@sentry/utils';
import { init } from '@sentry/electron/main';

init({
  dsn: '__DSN__',
  stackParser: createStackParser(nodeStackLineParser())
});
```

**Comment by armenzg** [comment]
Hi @timfish is there a demo app for Mac I can tweak?
Do you have a way to try with the same SDK version they're using?

Also, taking note that I believe this is lacking source maps, this is why the context line is full of snip values:
<img width="269" alt="image" src="https://github.com/getsentry/sentry-electron/assets/44410/d71ea1d3-8667-423e-a4ef-29ed809bf975">

**Comment by AbhiPrasad** [comment]
re-opened this as electron sdk change still needs to be made after we release JS SDK.

**Comment by timfish** [comment]
This should be fixed in v4.8.0 which should be released shortly!

**Comment by armenzg** [comment]
@timfish thank you for fixing this!
